spyder/utils/encoding.py

```diff
@@ -111,6 +113,17 @@ def get_coding(text):
             # sometimes we find a false encoding that can result in errors
             if codec in CODECS:
                 return codec
+
+    # Falback using chardet
+    if is_binary_string(text):

```

Comment:
This will only work for text that's not unicode, so why not remove from `CODECS` above all encodings that are not unicode and use chardet for non-unicode stuff?